app/models/katello/glue/pulp/rpm.rb

```diff
@@ -0,0 +1,38 @@
+module Katello
+  module Glue::Pulp::Rpm
+    extend ActiveSupport::Concern
+    include LazyAccessor
+
+    included do
+      lazy_accessor :pulp_facts, :initializer => :backend_data
+
+      lazy_accessor :description, :license, :buildhost, :vendor, :relativepath, :children, :checksumtype,
+                    :changelog, :group, :size, :url, :build_time, :group,
+                    :initializer => :pulp_facts
+    end
+
+    def requires
+      if pulp_facts['requires']
+        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
+      else
+        []
+      end
```

Comment:
I can see arguments both ways, and i'm really not sure why i debated this
so long, but in this particular case i just tend to prefer the if/else.  If
you had written the code the way you proposed I probably wouldn't have
given it a second thought.  If it were a longer function and it made sense
in context, a 'guard' clause may have felt more natural.

On Mon, Aug 3, 2015 at 7:26 PM, Eric D Helms <notifications@github.com>
wrote:

> In app/models/katello/glue/pulp/rpm.rb
> <https://github.com/Katello/katello/pull/5326#discussion_r36143224>:
>
> > +    include LazyAccessor
> > +
> > +    included do
> > +      lazy_accessor :pulp_facts, :initializer => :backend_data
> > +
> > +      lazy_accessor :description, :license, :buildhost, :vendor, :relativepath, :children, :checksumtype,
> > +                    :changelog, :group, :size, :url, :build_time, :group,
> > +                    :initializer => :pulp_facts
> > +    end
> > +
> > +    def requires
> > +      if pulp_facts['requires']
> > +        pulp_facts['requires'].map { |entry| Katello::Util::Package.format_requires(entry) }
> > +      else
> > +        []
> > +      end
>
> Likely the multiple returns comes more from JavaScript where, due to the
> language, it is typically encouraged to have only one return per function
> and use variable assignment instead. With Ruby's syntax structure I think,
> as of late, using guard clauses makes smaller, nice to read code that
> checks for the nil/empty/simply case and quickly handles it with the
> complex case following after (http://c2.com/cgi/wiki?GuardClause,
> https://www.youtube.com/watch?v=8bZh5LMaSmE). I am by no means requiring
> a change in this manner, more just using this as an opportunity to explore
> the thought between the two ways of tackling the design while I iterate
> test and review on the PR.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Katello/katello/pull/5326/files#r36143224>.
>